src/functions.js

```diff
@@ -67,7 +78,107 @@ function get_click_handler( item ) {
 	}
 }
 
-export {
-	get_click_handler,
-	extend_toolbar,
+export function addProviderFilter() {
+	const { providers } = AMF_DATA;
+
+	// Short circuit if we don't have providers
+	if ( ! providers?.length ) {
```

Comment:
As noted above this will be simpler to work around if we return `null` when there are no providers but if it's easier to pass an empty object then `.length` won't work here.

Reply:
Yes, somehow I kept forgetting that this is an object, as I only looked at code that used the values (so I was thinking `array` all the time).

Ensuring the `providers` property is either an object literal or `null` sounds good to me. 👍